evm/src/main/java/org/hyperledger/besu/evm/precompile/AbstractPrecompiledContract.java

```diff
@@ -53,4 +55,45 @@ protected GasCalculator gasCalculator() {
   public String getName() {
     return name;
   }
+
+  /** Default result caching to false unless otherwise set. */
+  protected static Boolean enableResultCaching = Boolean.FALSE;
+
+  /**
+   * Enable or disable precompile result caching.
+   *
+   * @param enablePrecompileCaching boolean indicating whether to cache precompile results
+   */
+  public static void setPrecompileCaching(final boolean enablePrecompileCaching) {
+    enableResultCaching = enablePrecompileCaching;
+  }
+
+  /** enum for precompile cache metric */
+  public enum CacheMetric {
+    /** a successful cache hit metric */
+    HIT,
```

Comment:
Can we make these lowercase, as they are used in the name of the metrics, may be implement a toString method.
Currently the name of besu metrics dans have uppercase names.

Reply:
Checking the way the metric is exposed, this is part of the metric attributes, so it is fine to keep it as it is.
We get the metric value with this kind of formula `besu_block_processing_precompile_cache_total{instance =~"dev-elc-bu-tk-mainnet-01", event="HIT", precompile_name="ECREC"}`.